app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js

```diff
@@ -116,6 +117,51 @@ window.GOVUK.analyticsGa4.analyticsModules = window.GOVUK.analyticsGa4.analytics
       }
     },
 
+    populateLinkPathParts: function (clickData) {
+      var href = clickData.url
+      var path = ''
+      if (this.hrefIsRelative(href)) {
+        path = href
+      } else if (this.isMailToLink(href)) {
+        path = href
+      } else {
+        path = href.replace(/^(http:||https:)?(\/\/)([^\/]*)/, '') // eslint-disable-line no-useless-escape
+      }
+
+      if (path === '/' || path.length === 0) {
+        return
+      }
+
+      /* Divide the length into hundreds, to determine how many string "parts" to make.
+      For example a 150 char string length divided by 100 would return 1.5,
+      so we round up, which returns 2, which makes sense as we would need one segment for chars 0-100, and another for chars 100-150 . */
+      var hundreds = Math.ceil(path.length / 100)
+
+      var LIMIT = 5
+
+      if (hundreds > LIMIT) {
+        hundreds = LIMIT // Limit the amount of data we send to 500 chars
+      }
+
+      clickData.link_path_parts = {}
+
+      for (var i = 0; i < LIMIT; i++) {
+        var stringIndex = (i + 1).toString() // Start the GTM index from 1 for analysts benefit
```

Comment:
If you want the loop to be 1 indexed you can do `for (var i = 1; i <= LIMIT; i++) {`

Reply:
Good idea :+1: